cypress/e2e/signUp.cy.js

```diff
@@ -1,12 +1,39 @@
 /// <reference types='cypress' />
 /// <reference types='../support' />
 
+import HomePageObject from '../support/pages/home.pageObject';
+import SignUpPageObject from '../support/pages/signUp.pageObject';
+
+const signUpPage = new SignUpPageObject();
+const homePage = new HomePageObject();
+
 describe('Sign Up page', () => {
-  before(() => {
+  let user;
+
+  beforeEach(() => {
+    cy.task('db:clear');
+    cy.task('generateUser').then((data) => {
+      user = data;
+    });
+
+    signUpPage.visit();
+  });
+
+  it('should provide an ability to sign up with valid data', () => {
+    signUpPage.typeUsername(user.username);
+    signUpPage.typeEmail(user.email);
+    signUpPage.typePassword(user.password);
+    signUpPage.clickSignUpButton();
 
+    homePage.assertHeaderContainUsername(user.username);
```

Comment:
The method `assertHeaderContainUsername` seems to have a grammatical error. It should likely be `assertHeaderContainsUsername` to correctly reflect the action being performed.